File: test/examples/bad-example.exs
# Elixir bad example anti-patterns
defmodule BadExample do
  def recursive_factorial(0), do: 1
  def recursive_factorial(n), do: n * recursive_factorial(n - 1)

  def unbounded_loop do
    Stream.repeatedly(fn -> :ok end) |> Enum.each(fn _ -> :ok end)
  end
end

File: test/examples/bad-example.zig
// Zig bad example anti-patterns
const std = @import("std");

pub fn recursiveFactorial(n: u64) u64 {
    if (n <= 1) return 1;
    return n * recursiveFactorial(n - 1);
}

pub fn unboundedLoop() void {
    while (true) {
        std.debug.print("looping\n", .{});
    }
}
